### ISSUE TITLE
`jsonProtocol` should report errors as nicely as before

### ISSUE THREAD

**Issue by SomiDivian** [prisma/prisma#18876]
As you can see, the `jsonProtocol` preview feature doesn't report errors as good as before. 

```prisma
model Wallet {
  id String @id @default(auto()) @map("_id") @db.ObjectId
  config WalletConfig[]
}

... TransferMethod enums

type TransferMethod {
  type     TransferMethodType
  token    String
  country  TransferMethodCountry
  currency TransferMethodCurrency
  email    String?
  status   TransferMethodStatus?
  meta     Json?
}

enum ConfigType {
  typeone
}

model WalletConfig {
  id String @id @default(auto()) @map("_id") @db.ObjectId

  type    ConfigType       @default(typeone)
  email   String
  token   String
  methods TransferMethod[]
  status  String?

  wallet   Wallet? @relation(fields: [walletId], references: [id])
  walletId String  @db.ObjectId
}
```

### Errors without `jsonProtocol`
- Wrong `method.country`
```cli
Argument country: Provided value 'country1' of type String on prisma.updateOneWalletConfig is not a TransferMethodCountry.
→ Possible values: TransferMethodCountry.Co1, ...
```
- Wrong `method.currency`
```cli
Argument currency: Provided value 'USD' of type String on prisma.updateOneWalletConfig is not a TransferMethodCurrency.
→ Possible values: TransferMethodCurrency.Cu1, ...
```

### Errors with `jsonProtocol`
```cli
→ 406 await prisma.walletConfig.update({
        where: {
          id: config_id
        },
        data: {
          type: "typeone", // same if we don't update this
          methods: {
            push: {
              type: "CORRECT_TYPE_FOR_SURE",
              country: "wrong_country", // this
              currency: "WRONG_CURRENCY", // or this is wrong
              token: "...",
              email: "...",
              meta: {
                metadata
              },
              status: "CORRECT_STATUS_FOR_SURE"
            },
      +     type: TransferMethodType
          }
        }
      })

Argument `type` is missing.
...
```

**Comment by SevInf** [comment]
Note: this is fixed by #18515

**Comment by SevInf** [comment]
Fix will be published as a part of Prisma 4.14.0.
